### PR TITLE
enhancement(Transitions): Mixin for duration and easing

### PR DESCRIPTION
**Pull request by hsarb**
Mixin to add easing and duration css properties.

Did not create a new build, as it is not currently being used.

Could use `platform` to detect current platform, but would need a list of platform os's that need to be supported.

### CODE REVIEW

File: src/mixins/transition.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { css } from 'styled-components';
+
+const isMobile =
+  typeof window.orientation !== 'undefined' || navigator.userAgent.indexOf('IEMobile') !== -1;
```
Comment (martahj) [inline src/mixins/transition.js]: [Q] Will this be true for tablets as well?

Comment (hsarb) [reply src/mixins/transition.js]: Should be yes.

Comment (brad-decker) [reply src/mixins/transition.js]: Can you walk me through this logic and how safe this is to use to determine if the device is mobile?

Comment (hsarb) [reply src/mixins/transition.js]: Sure. It's basically just looking to see if in the current browser window.oriententation exists. If so, the browser is most likely mobile.  I'm not sold on the safety of this as I was just using it for now as a quick win. I would prefer to use platform or navigator.userAgent to do the check, but would need a list of supported environments. 

Looks like my concerns are valid as it has been deprecated. 

https://developer.mozilla.org/en-US/docs/Web/API/Window/orientation

Going to update the PR with userAgent checks. Will just check for Android iOS and windows phone for now.

**Pull request merged**